### PR TITLE
Re-instate cube-env-map, fix it to work with 1.4.1 & add an example

### PR DESCRIPTION
**Pull request by diarmidmackenzie**
Based on discussions on A-Frame slack, our decision to deprecate cube-env-map in favour of A-Frame naive a-cubemap was premature.

In particular,` a-cubemap` is applied to an entity via the `material` component.  That `material` component doesn't play nicely with GLTF models, so the result is that `a-cubemap` can't be easily used to apply envMaps to GLTFs, which is a pretty mainline use case!

This has led me to look at re-instating cube-env-map.

This PR:
- adds cube-env-map back in
- adds an example with an A-Frame native geometry (sphere) and a GLTF (damaged helmet)
- fixes a bug where cube-env-map never worked with native geometries.
- fixes compatibility with A-Frame 1.3.0 and 1.4.0 by retiring RGBFormat (deprecated) in favour of RGBAFormat.

### CODE REVIEW

File: src/misc/cube-env-map.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+  init: function () {
+    const data = this.data;
+
+    this.texture = new THREE.CubeTextureLoader().load([
+      data.path + 'posx.' + data.extension, data.path + 'negx.' + data.extension,
+      data.path + 'posy.' + data.extension, data.path + 'negy.' + data.extension,
+      data.path + 'posz.' + data.extension, data.path + 'negz.' + data.extension
+    ]);
+    this.texture.format = THREE.RGBAFormat;
```
Comment (diarmidmackenzie) [inline src/misc/cube-env-map.js:71]: This is one of the 2 bug fixes in this component.  Previously format was specifiable on the schema as RGB or RGBA, with RGB the default.

File: src/misc/cube-env-map.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    this.object3dsetHandler = () => {
+      const mesh = this.el.getObject3D('mesh');
+      const data = this.data;
+      applyEnvMap(mesh, data.materials, this.texture, data.reflectivity);
+    };
+
+    this.object3dsetHandler();
```
Comment (diarmidmackenzie) [inline src/misc/cube-env-map.js:79]: Other bug fix is to call `this.object3dsetHandler()` immediately - needed for the case where we are working with a antive geometry, where we won't get a subsequent object3dset event.

File: src/misc/cube-env-map.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
+/**
```
Comment (diarmidmackenzie) [inline src/misc/cube-env-map.js:1]: This module is re-instated as it was previously, except for a couple of small bug fixes commented inline

**Comment by diarmidmackenzie** [comment]
Here's what the example looks like:

image

**Review by vincentfretin — APPROVED** [review]
Alright, at quick glance it looks good to me. Let's merge.

**Pull request merged**